src/overlays/actors/ovl_Dm_Tsg/z_dm_tsg.c

```diff
@@ -28,14 +28,66 @@ ActorInit Dm_Tsg_InitVars = {
     /**/ DmTsg_Draw,
 };
 
-#endif
+void DmTsg_Init(Actor* thisx, PlayState* play) {
+    DmTsg* this = THIS;
+    s32 i;
 
-extern UNK_TYPE D_06011458;
+    if (gSaveContext.save.entrance == ENTRANCE(OPENING_DUNGEON, 0)) {
+        Actor_SetScale(&this->actor, 0.1f);
+        for (i = 0; i < DMTSG_UNK_SIZE; i++) {
+            this->unk_78C[i] = (Rand_ZeroOne() * 65535.0f);
```

Comment:
```suggestion
            this->unk_78C[i] = Rand_ZeroOne() * 65535.0f;
```
Small nit

Reply:
Since `unk_78C` is a rotation, `65535.0f` can be in hex too